server/api.go

```diff
@@ -12,22 +14,34 @@ import (
 	"github.com/mattermost/mattermost-server/v6/model"
 )
 
+const (
+	KeyClusterMutex = "mutex_exporter"
+)
+
 // Handler encapsulates the context necessary for the channel export API.
 type Handler struct {
 	client            *pluginapi.Wrapper
 	makePostsIterator func(*model.Channel, bool) PostIterator
+	clusterMutex      pluginapi.ClusterMutex
 }
 
 // registerAPI registers the API against the given router.
-func registerAPI(router *mux.Router, client *pluginapi.Wrapper, makePostsIterator func(*model.Channel, bool) PostIterator) {
+func registerAPI(router *mux.Router, client *pluginapi.Wrapper, makePostsIterator func(*model.Channel, bool) PostIterator) error {
+	clusterMutex, err := client.Cluster.NewMutex(KeyClusterMutex)
```

Comment:
Why not passing [`p` or `p.clusterMutex`](https://github.com/mattermost/mattermost-plugin-channel-export/pull/43/files#diff-43f9ea431ff164da5166ef1e23e4a37f80777429b6803d8303b07b555cb1c1cfR63) to `registerAPI`?

Reply:
I would see that as an unnecessary change.  The amount of QA we'll get for this release is minimal, so I would like to change as little as possible.